packages/uploadthing/src/client.ts

```diff
@@ -312,3 +260,137 @@ export const generateClientDropzoneAccept = (fileTypes: string[]) => {
 };
 
 export { resolveMaybeUrlArg };
+
+async function uploadMultipart(
+  file: File,
+  presigned: MPUResponse,
+  opts: {
+    reportEventToUT: ReturnType<typeof createUTReporter>;
+    onUploadProgress?: UploadFilesOptions<any, any>["onUploadProgress"];
+  },
+) {
+  let etags: { tag: string; partNumber: number }[];
+  let uploadedBytes = 0;
+
+  try {
+    etags = await Promise.all(
+      presigned.presignedUrls.map(async (url, index) => {
```

Comment:
Any reason we don't destructure this higher?

Reply:
no more than i didn't see a need and don't like destructuring unless i feel like i need to. if you think it's cleaner with it destructured happy to have it

Reply:
Maybe rename field on infra to be just `url` and `urls`?

Reply:
`data` ?

Reply:
>Maybe rename field on infra to be just url and urls?

would be breaking, no?

Reply:
Can do it just for this version?

Reply:
e.g. like this https://github.com/pingdotgg/uploadthing-infra/pull/316/commits/57f5b2a0d5c715dd2eb028b23df78c993582bee0